src/components/token/Tokens.js

```diff
@@ -20,20 +23,155 @@ class Tokens extends React.Component {
+        this.setState({ isSearchLoading: true });
+        //When search button is clicked, results return to the first page
+        this.resetPageSearchAfter();
+        const tokens = await this.getTokens([]);
+
+        this.setState({ isSearchLoading: false, page: 1, tokens: tokens.hits, hasAfter: tokens.has_next, hasBefore: false });
+    }
+
+    /**
+     * Updates searchText state value when input field is changed
+     *
+     * @param {*} event
+     */
     onSearchTextChanged = (event) => {
-        this.setState({ searchText: event.target.value })
+        this.setState({ searchText: event.target.value });
+    }
+
+    /**
+     * Checks if enter button is pressed. If so, treat as a button click on search icon
+     *
+     * @param {*} event
+     */
+    onSearchTextKeyUp = (event) => {
+        if (event.key === 'Enter') {
+            this.onSearchButtonClicked();
+        }
+    }
+
+    /**
+     * Process events when next page is requested by user
+     *
+     * @param {*} event
+     */
+    nextPageClicked = async (event) => {
+        this.setState({ calculatingPage: true });
+        const nextPage = this.state.page + 1;
+        let searchAfter = get(find(this.state.pageSearchAfter, { page: nextPage }), 'searchAfter', []);
+
+        //Calculate searchAfter of next page if not already calculated
+        if (isEmpty(searchAfter)) {
+            const lastCurrentTokenSort = get(last(this.state.tokens), 'sort', []);
+
+            const newEntry = {
+                page: nextPage,
+                searchAfter: lastCurrentTokenSort
+            }
+
+            this.setState({ pageSearchAfter: [...this.state.pageSearchAfter, newEntry] });
+            searchAfter = lastCurrentTokenSort;
+        }
+
+        const tokens = await this.getTokens(searchAfter);
+        this.setState({ tokens: tokens.hits, hasAfter: tokens.has_next, hasBefore: true, page: nextPage, calculatingPage: false });
+    }
+
+    /**
+     * Process events when previous page is requested by user
+     *
+     * @param {*} event
+     */
+    previousPageClicked = async (event) => {
+        this.setState({ calculatingPage: true });
+        const previousPage = this.state.page - 1;
+        const searchAfter = get(find(this.state.pageSearchAfter, { page: previousPage }), 'searchAfter', []);
+
+        const tokens = await this.getTokens(searchAfter);
+        this.setState({ tokens: tokens.hits, hasAfter: true, hasBefore: previousPage === 1 ? false : true, page: previousPage, calculatingPage: false });
+    }
+
+    /**
+     * Process table header click. This indicates that user wants data to be sorted by a determined field
+     *
+     * @param {*} event
+     * @param {*} header
+     */
+    tableHeaderClicked = async (event, header) => {
+        if (header === this.state.sortBy) {
+            await this.setState({ order: this.state.order === "asc" ? "desc" : "asc" });
+        } else {
+            await this.setState({ sortBy: header });
```

Comment:
Maybe set `order` to `asc` here

Reply:
Good catch. I will change on the next commit